server/src/main/java/org/elasticsearch/script/ScriptContextStats.java

```diff
@@ -43,6 +59,68 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(compilations);
         out.writeVLong(cacheEvictions);
         out.writeVLong(compilationLimitTriggered);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeList(List.of(compilationsHistory, cacheEvictionsHistory));
+        }
+    }
+
+    public static class TimeSeries implements Writeable, ToXContentFragment {
+        public final long five;
+        public final long fifteen;
+        public final long day;
```

Comment:
I would prefer these names carry the units. While it may be intuitive to us, I could see five and fifteen being confused with hours since the next increment is day. Maybe something like `fiveMinutes`, `fifteenMinutes`, `oneDay`.

Reply:
Done.